src/main/java/core/basesyntax/Figures.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figures {
+    private static Random random = new Random();
+    double area;
+    Color color;
+
+    public abstract void displayInfo();
+
+    public abstract void draw();
+
+    public Color getColor() {
+        return color;
+    }
+
+    public double getArea() {
+        return area;
+    }
+
+    public static Figures getRandomFigure() {
```

Comment:
With this logic your abstract class depends on the impl and it's a bad practice, move this logic to separate class, e.g. "FigureGenerator". This way your code will follow single responsibility principle